app/src/main/java/com/example/workoutapp/ProfileActivity.kt

```diff
@@ -46,8 +47,10 @@ class ProfileActivity : ComponentActivity() {
                             weight = "57kg",
                             bmi = "21.8",
                             goals = goals,
-                            workouts = workouts
                         )
+                        for (workout in workouts) {
```

Comment:
Pulling this out of your compose functions improves performance since this will never have to be recomposed - https://developer.android.com/jetpack/compose/performance/bestpractices